Azure-Samples/chat-with-your-data-solution-accelerator
# feat:  PostgreSQL implementation for Chat History and Vectorization

## Purpose

* This pull request introduces PostgreSQL as a core feature for CWYD, allowing the platform to utilize PostgreSQL for chat history, search indexing, and vector-based similarity queries. It replaces the default database configuration with PostgreSQL and ensures all related infrastructure and backend components are updated accordingly.

## Does this introduce a breaking change?





## How to Test
* Get the code


* Test the code

1. Set up PostgreSQL as the default database:
    - Configure the necessary environment variables:
      ```
      export DATABASE_TYPE=PostgreSQL
      export POSTGRES_URI=your_postgres_uri
      export POSTGRES_USER=your_postgres_user
      export POSTGRES_PASSWORD=your_postgres_password
      ```
    - Ensure `pgvector` extension is installed in your PostgreSQL database:
      ```
      CREATE EXTENSION vector;
      ```
2. Run the infrastructure deployment script to create the necessary tables
3. Test chat history and search functionalities by interacting with the CWYD platform.

## What to Check
Verify that the following are valid:
* Chat history is stored and retrieved successfully from PostgreSQL.
* Search index creation and vector similarity queries work as expected using PostgreSQL.
* Advanced image processing and integrated vectorization features are disabled when `DATABASE_TYPE=PostgreSQL`.
* Switching back to CosmosDB (`DATABASE_TYPE=CosmosDB`) retains existing functionality.

## Other Information

* Ensure that PostgreSQL credentials are securely managed in the environment or using Azure Key Vault.
* For any issues during testing or deployment, check logs for connection and query errors.
* This change replaces CosmosDB as the default database, which may affect deployments relying on previous configurations.
